src/Arcus.Security.Providers.DockerSecrets/DockerSecretsSecretProvider.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Arcus.Security.Core;
+using GuardNet;
+using Microsoft.Extensions.Configuration.KeyPerFile;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileProviders;
+
+namespace Arcus.Security.Providers.DockerSecrets
+{
+    /// <summary>
+    /// Represents an <see cref="ISecretProvider" /> that provides access to the Docker secrets mounted into the Docker container as files.
+    /// </summary>
+    public class DockerSecretsSecretProvider : ISecretProvider
+    {
+        private readonly KeyPerFileConfigurationProvider _provider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DockerSecretsSecretProvider"/> class.
+        /// </summary>
+        /// <param name="secretsDirectoryPath">The path inside the docker container where the secrets are located.</param>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="secretsDirectoryPath"/> is <c>null</c>or an empty string.</exception>
+        public DockerSecretsSecretProvider(string secretsDirectoryPath)
+        {
+            Guard.NotNullOrWhitespace(secretsDirectoryPath, nameof(secretsDirectoryPath));
+
```

Comment:
Should we here check if the reference to the secret directory path points to a existing directory? Or should we let the remaining code do that. I'm only thinking that the exception thrown there will maybe more obscure and will not include the information that we're in the `DockerSecretProvider`.

Reply:
True.  If added a check and throw an exception in the DockerSecretsProvider if the directory doesn't exist.

However, I'm still not convinced if throwing an exception (and using Optional=false) is the approach that we need to take.  See also this comment: https://github.com/arcus-azure/arcus.security/pull/171#discussion_r502911108

Reply:
Yes, lets talk about that in here #200.